src/pages/Polling.js

```diff
@@ -112,140 +109,414 @@ const DetailsCardText = styled.p`
-              { name: 'Unique voters', value: '1041' }
-            ].map((item, i) => (
-              <DetailsCardItem key={i} {...item} />
-            ))}
-
-            <CardTitle>Vote breakdown</CardTitle>
-            {[
-              { name: '17.5%', value: '170,324 MKR (82%)' },
-              { name: '19.5%', value: '20,520 MKR (12%)' },
-              { name: 'No change', value: '100 MKR (0.5%)' },
-              { name: '14.5%', value: '0 MKR' },
-              { name: '15.5%', value: '0 MKR' },
-              { name: '16.5%', value: '0 MKR' }
-            ].map((item, i) => (
-              <DetailsCardItem key={i} {...item} />
-            ))}
-          </DetailsPanelCard>
-        </RightPanels>
-      </ContentWrapper>
-    </RiseUp>
-  );
+            >
+              Withdraw Vote
+            </Blue>
+          </Fragment>
+        )}
+      </VoteStatusText>
+    );
+  else
+    return (
+      <VoteStatusText>
+        <Black>{active ? 'Not currently voting' : 'You did not vote'}</Black>
+      </VoteStatusText>
+    );
+};
+
+class VotingPanel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedOption: null
+    };
+  }
+
+  onDropdownSelect = (value, index) => {
+    const selectedOptionId = parseInt(index) + 1;
+    this.setState({
+      selectedOption: value,
+      selectedOptionId
+    });
+  };
+
+  formatOptions = options => {
+    const displayOptions = [...options];
+    // Index 0 is expected to always be '0: abstain' but we don't want to display it here.
+    displayOptions.shift();
+    return displayOptions;
+  };
+
+  render() {
+    const {
+      poll,
+      activeAccount,
+      optionVotingFor,
+      modalOpen,
+      totalVotes
+    } = this.props;
+    const { pollId, options } = poll;
+    const { selectedOption, selectedOptionId } = this.state;
+
+    return (
+      <React.Fragment>
+        <VoteSelection>
+          <Dropdown
+            color="green"
+            items={this.formatOptions(options)}
+            renderItem={item => (
+              <DropdownText color="green">{item}</DropdownText>
+            )}
+            renderRowItem={item => <DropdownText>{item}</DropdownText>}
+            value={selectedOption || optionVotingFor || 'Please choose...'}
+            onSelect={this.onDropdownSelect}
+            emptyMsg="Not available"
+          />
+          <VoteButton
+            bgColor="green"
+            color="white"
+            hoverColor="white"
+            width="135px"
+            disabled={
+              !poll.active ||
+              !activeAccount ||
+              selectedOptionId === undefined ||
+              selectedOption === optionVotingFor
+            }
+            onClick={() =>
+              modalOpen(PollingVote, {
+                poll: {
+                  pollId,
+                  selectedOption,
+                  selectedOptionId,
+                  totalVotes
```

Comment:
hmm i think we want `totalVotes` to be the total votes for that specific option. otherwise "Current vote" in the modal represents the total mkr voting in that poll which can lead to some unexpected behavior. ex create a new poll, vote option `a` w/ 100 mkr, see "Current vote" as 0 mkr & "After vote cast" as 100 mkr, change vote to option `b`, see "Current vote" as 100 mkr & "After vote cast" as 200 mkr.

Reply:
That makes sense, I can refactor `voteBreakdown` to pass the `mkrSupport` for that option to this component.